src/battle_ai_main.c

```diff
@@ -2868,6 +2868,14 @@ static s32 AI_DoubleBattle(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
         {
             switch (atkPartnerAbility)
             {
+            case ABILITY_ANGER_POINT:
+                if (gMovesInfo[move].alwaysCriticalHit == TRUE 
+                    && BattlerStatCanRise(battlerAtkPartner, atkPartnerAbility, STAT_ATK)
+                    && !CanIndexMoveFaintTarget(battlerAtk, battlerAtkPartner, AI_THINKING_STRUCT->movesetIndex, 1))
```

Comment:
it would probably be useful to also check speed. Also delete the other check above since it isn't needed anymore

Reply:
oh, right. my bad

Reply:
Added the speed check, should be resolved now.